rest_auth_toolkit/views.py

```diff
@@ -48,19 +65,71 @@ def post(self, request):
         """
         deserializer = self.get_serializer(data=request.data)
         deserializer.is_valid(raise_exception=True)
-        user = deserializer.save()
 
-        if self.email_confirmation_class is None:
-            raise MissingSetting('email_confirmation_string')
+        confirm_email = get_setting('email_confirmation_send_email', True)
 
-        confirmation = self.email_confirmation_class.objects.create(user=user)
-        if get_setting('email_confirmation_send_email', True):
+        if not confirm_email:
+            deserializer.save(is_active=True)
+        else:
+            user = deserializer.save()
+
+            if self.email_confirmation_class is None:
+                raise MissingSetting('email_confirmation_class')
```

Comment:
This check could be moved to SignupView.as_view, so that the problem is visible when the site starts rather than during the first request.

Reply:
Better: https://github.com/caravancoop/rest-framework-auth-toolkit/issues/52